test/IntegrationTests/MSTest.Acceptance.IntegrationTests/AssemblyAndClassInitializeTimeout.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests;
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests.Helpers;
+
+namespace MSTest.Acceptance.IntegrationTests;
+
+[TestGroup]
+public class AssemblyAndClassInitializeTimeout : AcceptanceTestBase
+{
+    private readonly string _assetName = "AssemblyAndClassInitializeTimeout";
+    private readonly AcceptanceFixture _acceptanceFixture;
+
+    public AssemblyAndClassInitializeTimeout(ITestExecutionContext testExecutionContext, AcceptanceFixture acceptanceFixture)
+        : base(testExecutionContext)
+    {
+        _acceptanceFixture = acceptanceFixture;
+    }
+
+    [ArgumentsProvider(nameof(GetBuildMatrixTfmBuildConfiguration))]
+    public async Task AssemblyInitialize_Timeout_Should_Stop_Execution(string tfm, BuildConfiguration buildConfiguration)
+    {
+        using TestAsset testAsset = await TestAsset.GenerateAssetAsync(_assetName, SourceCode
+            .PatchCodeWithReplace("$TargetFramework$", tfm)
+            .PatchCodeWithReplace("$MSTestVersion$", MSTestVersion), addPublicFeeds: true);
+        DotnetMuxerResult result = await DotnetCli.RunAsync($"build -nodeReuse:false {testAsset.TargetAssetPath} -c {buildConfiguration}", _acceptanceFixture.NuGetGlobalPackagesFolder.Path);
+        var testHost = TestHost.LocateFrom(testAsset.TargetAssetPath, _assetName, tfm, buildConfiguration: buildConfiguration);
+
+        foreach (string initMethodToTimeout in new[] { "TIMEOUT_CLASSINIT", "TIMEOUT_ASSEMBLYINIT", "TIMEOUT_BASE_CLASSINIT" })
```

Comment:
I am wondering if the resource creation should be done once only in some init and then you would have 3 tests that simply executes the assembly with different env var set. This would simplify understand failed case and make test reading easier.